update from #861 in main-md (#1332)
substrate-developer-hub/substrate-docs: content/md/en/docs/tutorials/work-with-pallets/use-macros-in-a-custom-pallet.md

   All of the pallets used in a runtime must be set to compile with the `no_std` features.

> 1. Add the following skeleton code that includes a few minimal [macros for FRAME V2](https://paritytech.github.io/substrate/master/frame_support/attr.pallet.html):

Reviewer: ```suggestion
1. Add the following code that includes a minimal skeleton for a FRAME pallet and its required [macros](https://paritytech.github.io/substrate/master/frame_support/attr.pallet.html):
```
FRAME is FRAME unless we explicitly need to refer to a previous version :)

   ```rust

>    //! Substrate Proof-of-Existence Pallet
>    #![cfg_attr(not(feature = "std"), no_std)]
> 

Reviewer: ```suggestion
```
This is in step 1 already.

Author: I see the style Lisa is using is _super extra_ explicit in steps... think we should pull this into it's own step? IMHO no.

   #[frame_support::pallet]
   pub mod pallet {
   	use frame_support::{pallet_prelude::*, storage::bounded_vec::BoundedVec};
   	use frame_system::pallet_prelude::*;

   	// Define the pallet struct placeholder, various pallet function are implemented on it.
   	#[pallet::pallet]
   	#[pallet::generate_store(pub(super) trait Store)]
   	pub struct Pallet<T>(_);

   	// TODO: Update the `Config` block below
   	/// Configure the pallet by specifying the parameters and types on which it depends.
   	#[pallet::config]
   	pub trait Config: frame_system::Config {
   		/// Because this pallet emits events, it depends on the runtime's definition of an event.
   		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
   	}

   	// TODO: Update the `event` block below
   	#[pallet::event]
   	#[pallet::generate_deposit(pub(super) fn deposit_event)]
   	pub enum Event<T: Config> {
   		/// Event documentation should end with an array that provides descriptive names for event
   		/// parameters. [something, who]
   		SomethingHappened(),
   	}
   	// TODO: add #[pallet::error] block

   	// TODO: add #[pallet::storage] block

   	// TODO: add #[pallet::call] block

>    }

Author: @sacha-l this was intentional, not formatting :grimacing:

Author: specifically, this code ***compiles*** - so we need `Event` defined, though we replace it in a moment

   cargo check -p node-template-runtime
   ```

> > **_Something not working?_**

Reviewer 2: ```suggestion
```
Please move these solution links to the end. Including it here implies that you *expect* it not to work. That means it is a bad tutorial. It's find to provide a solution reference, but we've just had them verify that it compiles, so this seems weird.
